src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -612,10 +612,7 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
         }
       }
 
-      val companionObjectsCount = typeSpecs.count {
-        // Replace this with isCompanion check after merged
-        it.kind is Object && COMPANION in it.kind.modifiers
-      }
+      val companionObjectsCount = typeSpecs.count { it.kind.let { it is Object && it.isCompanion } }
```

Comment:
Is the nested `let` needed? I would rather it omitted. We should ban nested use of `it`s in the style guide.

Reply:
Technically, no. I can remove it here but I personally don't think it should be blanket banned in a style guide :|

Reply:
7fce736